utils/crates-io/src/handler.rs

```diff
@@ -0,0 +1,215 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Handlers for patching manifests.
+
+use crate::Manifest;
+use anyhow::Result;
+use cargo_metadata::Package;
+use toml_edit::Document;
+
+/// Patch specified manifest by provided name.
+pub fn patch(pkg: &Package) -> Result<Manifest> {
+    let mut manifest = Manifest::new(pkg)?;
+    let doc = &mut manifest.manifest;
+
+    match manifest.name.as_str() {
+        "gear-core-processor" => core_processor::patch(doc),
+        "gear-runtime-interface" => runtime_interface::patch(doc),
+        "gear-sandbox" => sandbox::patch(doc),
+        "gear-sandbox-host" => sandbox_host::patch(doc),
+        "gmeta" => gmeta::patch(doc),
+        "gmeta-codegen" => gmeta_codegen::patch(doc),
+        _ => {}
+    }
+
+    Ok(manifest)
+}
+
+/// Patch package alias.
+pub fn patch_alias(index: &mut Vec<&str>) {
+    for (package, alias) in crate::PACKAGE_ALIAS {
+        if index.contains(&package) {
+            index.push(alias);
+        }
+    }
+}
+
+/// Patch the workspace manifest.
+pub fn patch_workspace(name: &str, table: &mut toml_edit::InlineTable) {
+    match name {
+        "core-processor" | "gear-core-processor" => core_processor::patch_workspace(name, table),
+        sub if sub.starts_with("sp-") => substrate::patch_workspace(name, table),
+        _ => {}
+    }
+}
+
+// Trim the version of dev dependency.
+//
+// issue: https://github.com/rust-lang/cargo/issues/4242
+fn trim_dev_dep(name: &str, manifest: &mut Document) {
+    if let Some(dep) = manifest["dev-dependencies"][name].as_inline_table_mut() {
+        dep.remove("workspace");
+        dep.insert("version", "~1".into());
+    }
+
+    if let Some(dep) = manifest["dev-dependencies"][name].as_table_like_mut() {
+        dep.remove("workspace");
+        dep.insert("version", toml_edit::value("~1"));
+    }
+}
+
+/// gear-core-processor handler.
+mod core_processor {
+    use toml_edit::{Document, InlineTable};
+
+    /// Rename core processor related package in the
+    /// manifest of workspace since `gear-core-processor`
+    /// has been published by others.
+    pub fn patch_workspace(name: &str, table: &mut InlineTable) {
+        match name {
```

Comment:
why do we remove if exist and add if not?

Reply:
it's for pointing the both alias to https://crates.io/crates/core-processor, see a58644f